### PR TITLE
Python verbose_eval extension

### PR DESCRIPTION
**Pull request by JohanManders**
Printing the evaluation metric every boosting stage is very handy, but if you have a lot of boosting rounds it will become a long cluttered list and can for example slow down Chrome when using iPython.

The verbose_eval extension will make it possible to print the evaluation metric **not every boosting round**, but every **x** boosting rounds by assigning it an ```integer``` instead of ```True```. It will always print the last boosting round and when given ```early_stopping_rounds``` it will print the best iteration.

The old functionality stays the same, this will only give you extra functionality. There is one exception. In current master the best iteration was printed, even if you explicitly use ```verbose_eval=False```. After this update, it will not anymore. If you say you don't want to print it, it should not be printed.

The docstring reflects the changes.

**Examples**

Using ```verbose_eval=4``` without using ```early_stopping_rounds```:
```python
import xgboost as xgb
import numpy as np
from sklearn.datasets import make_hastie_10_2

X, y = make_hastie_10_2(n_samples=500, random_state=42)

labels, y = np.unique(y, return_inverse=True)

X_train, X_test = X[:400], X[400:]
y_train, y_test = y[:400], y[400:]

dtrain = xgb.DMatrix(X_train, y_train)
dtest = xgb.DMatrix(X_test, y_test)

param = [('max_depth', 2),
         ('objective', 'binary:logistic'),
         ('learning_rate', 0.5),
         ('eval_metric', 'logloss')]

watchlist = [(dtrain, 'train'), (dtest, 'eval')]
num_round = 10
evals_result = {}

bst = xgb.train(param, dtrain, num_round, watchlist, verbose_eval=4)
```
Will only print 3 lines and the last line:
```
[0]	train-logloss:0.647351	eval-logloss:0.656927
[4]	train-logloss:0.514507	eval-logloss:0.637059
[8]	train-logloss:0.427117	eval-logloss:0.597834
[9]	train-logloss:0.409596	eval-logloss:0.596313
```
Using ```verbose_eval=4``` with ```early_stopping_rounds=2```:
```python
bst = xgb.train(param, dtrain, num_round, watchlist, early_stopping_rounds=2, verbose_eval=4)
```
Will print 2 lines and 1 line with the best iteration:
```
Will train until eval error hasn't decreased in 2 rounds.
[0]	train-logloss:0.647351	eval-logloss:0.656927
[4]	train-logloss:0.514507	eval-logloss:0.637059
Stopping. Best iteration:
[3]	train-logloss:0.544489	eval-logloss:0.622650
```
When using ```verbose_eval=True```:
```python
bst = xgb.train(param, dtrain, num_round, watchlist, verbose_eval=4)
```
Will print all lines, like in current master:
```
[0]	train-logloss:0.647350	eval-logloss:0.656927
[1]	train-logloss:0.605819	eval-logloss:0.654618
[2]	train-logloss:0.575058	eval-logloss:0.640290
[3]	train-logloss:0.544489	eval-logloss:0.622650
[4]	train-logloss:0.514507	eval-logloss:0.637059
[5]	train-logloss:0.491549	eval-logloss:0.628161
[6]	train-logloss:0.469362	eval-logloss:0.615877
[7]	train-logloss:0.447411	eval-logloss:0.605581
[8]	train-logloss:0.427117	eval-logloss:0.597834
[9]	train-logloss:0.409596	eval-logloss:0.596313
```

### CODE REVIEW

**Comment by terrytangyuan** [comment]
@JohanManders Thanks. Could you squash?

**Comment by JohanManders** [comment]
@terrytangyuan Squashed!

**Comment by Far0n** [comment]
coll stuff! :)

**Comment by JohanManders** [comment]
Thanks @Far0n & @terrytangyuan
